src/backend/aspen/app/views/tests/test_samples_view.py

```diff
@@ -20,8 +22,24 @@ def test_samples_create_view_pass_no_public_id(
 ):
     group = group_factory()
     user = user_factory(group)
+    location = location_factory(
+        "North America", "USA", "California", "Santa Barbara County"
+    )
     session.add(group)
+    session.add(location)
     session.commit()
+    location = (
+        session.query(Location)
+        .filter(
+            and_(
+                Location.region == "North America",
+                Location.country == "USA",
+                Location.division == "California",
+                Location.location == "Santa Barbara County",
+            )
+        )
+        .one()
+    )
```

Comment:
Why create the location and then query it back out of the db? Is this so we can get its id?

Reply:
Yep

Reply:
gross. Is there some sort of SA flag we can twiddle to make it smarter?

Reply:
weird thing -- I checked out your branch and deleted this query and the tests still pass! I think the session.commit() in there takes care of updating the id!

Reply:
weirdddd